src/Spatial/Euclidean/Circle2D.cs

```diff
@@ -131,6 +133,66 @@ public static Circle2D FromPoints(Point2D pointA, Point2D pointB, Point2D pointC
             return new Circle2D(center, radius);            
         }
 
+        /// <summary>
+        /// Returns intersection a point2D array if this circle and the given line have the intersections
+        /// </summary>
+        /// <param name="line">the given line</param>
+        /// <returns>intersections as a Point2D Array, depending on the count.</returns>
+        public Point2D[] IntersectWith(Line2D line)
+        {
+            var ts = this.findParameterTs(line);
+            var result = ts
+                .Select(t => line.StartPoint + t * line.Direction)
+                .ToArray();
+            return result;
+        }
+
+        private double[] findParameterTs(Line2D line)
+        {
+            // These 2 equations in vector form can be described
+            // (p-cc)^2=r^2 (eq1)
+            // p=s+t*d     (eq2)
+            // , where p is the point on the line and/or circle,
+            // cc is the center of the circle,
+            // r is the radius of the circle,
+            // s is the starting point of the line,
+            // t is the parameter and
+            // d is the line direction.
+            // Substituting (eq2) into (eq1) yields:
+            // ((s+t*d)-cc)^2=r^2 (eq3)
+            // (eq3) reduces to the following quadratic equation: a*t^2 + b*t + c==0
+
+            var cc = this.Center.ToVector2D(); //center of circle
+            var s = line.StartPoint.ToVector2D();
+            var d = line.Direction;
+            var r = this.Radius;
+
+            var a = 1d;
+            var b = 2 * (s.DotProduct(d) - d.DotProduct(cc));
+            var c = (s - cc).DotProduct(s - cc) - r * r;
+
+            var soluions = FindRoots.Polynomial(new[] { c, b, a });
```

Comment:
`solutions`

Reply:
Thanks for the comment.
I'm on finxing this typo.

Reply:
done in the commit https://github.com/f-frhs/mathnet-spatial/commit/9e33c1b1daff669f22e59ed4051cf230c21f66f6